torch/nn/modules/module.py

```diff
@@ -26,6 +26,24 @@ def __repr__(self):
     __str__ = __repr__
 
 
+class _HookWrapper:
+    def __init__(self, hook: Callable[..., None], with_kwargs=False) -> None:
+        self.hook = hook
+        self.with_kwargs = with_kwargs
+
+    def run(self, module, input, kwargs):
+        if self.with_kwargs:
+            return self.hook(module, input, kwargs)
+        else:
+            return self.hook(module, input)
```

Comment:
You will need to fix the return type here to make sure both the updated input and kwargs are returned.